### PR TITLE
Qute: allow extensions to register additional template roots

### PR DESCRIPTION
**Pull request by mkouba**
- via TemplateRootBuildItem
- the default root is "templates"

### CODE REVIEW

**Comment by ia3andy** [comment]
@FroMage thanks to this with web-bundler, we will be able to have all the web stuff in `web/**` (with Web-Bundler + Renarde too)

**Comment by mkouba** [comment]
I found a small bug though, will update the PR shortly.

**Comment by ia3andy** [comment]
@fbricon @angelozerr you can already mark `web/templates` as a qute templates dir when using the `web-bundler`

**Comment by angelozerr** [comment]
Thanks @ia3andy for your feedback

**Comment by FroMage** [comment]
+1

**Comment by mkouba** [comment]
The PR should be ready now..

File: extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateRootBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public String getPath() {
+        return path;
+    }
+
+    static String normalize(String path) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        if (path.endsWith("/")) {
+            path = path.substring(0, path.length());
```
Comment (ia3andy) [inline extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/TemplateRootBuildItem.java:30]:
```suggestion
            path = path.substring(0, path.length() - 1);
```
right?

**Review by ia3andy — APPROVED** [review]
I believe there is a small bug on the replace. Beside this I have tested on the web-bundler and it's perfect!

**Pull request merged**